src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
 
             let mut lnd = handler.client;
 
-            let headers = Headers(req.headers().to_owned());
-            let query = {
-                let uri = req.uri();
-                if let Some(query) = uri.query() {
-                    Some(&query.to_owned());
-                }
-                None
-            };
-            let body = req.into_body();
-            let bytes = hyper::body::to_bytes(body).await?;
-            dbg!(&bytes); // this is correct by my accounts
-            let reader = &*bytes;
-            let proposal = UncheckedProposal::from_request(reader, query, headers).unwrap();
+            // extract original PSBT from HTTP request
+            // TODO: `UncheckedProposal` is not really the proposal, but the original PSBT? Is the naming wrong?
```

Comment:
UncheckedProposal is the OriginalPSBT + the sender's paramaters. It is a PayJoin proposal

Reply:
I mean the `UncheckedProposal` struct is badly named, because the proposed PSBT is what is returned to the receiver.